### DISCUSSION TITLE
MEGA-BAS RS485 Rpi passthrough

### DISCUSSION THREAD

**Discussion by orrious** [tagyoureit/nodejs-poolController#532] [RS485] [answered]
I made the leap today from my own hardware to REM with MEGA-BAS .  The only thing I wasn't able to test prior to switching was RS-485 and now having problems with getting it working from RPi through the MEGA-BAS.

I've disabled bluetooth and logins on all the RPi UARTS.  I'm not able to see any traffic on the serial ports when I `screen /dev/ttyAMA0 9600` or `screen /dev/ttyS0`

Jumpers are correct on the MEGA-BAS and in REM I have the RS485 port in passthrough

What am I missing?

**Comment by tagyoureit** [comment]
Make sure that your serial port is enabled and the serial console is disabled.  https://github.com/rstrouse/relayEquipmentManager/wiki/Sequent-Microsystems-Boards#rs-485-tab

It should start being chatty on `/dev/ttyS0`.

**Reply by orrious** [reply]
I followed these to setup it up.  Still no joy.  I've installed my old USB-RS485 adapter and have it working that way for now.  I also left the MEGA-BAS RS485 on it as well so I could sniff traffic live traffic.

That said, I'm unable to open /dev/ttyS0.  It baffles me as I've turned off bluetooth, disabled serial login, enabled the serial port, etc.  

/dev/ttyAMA0 does see traffic, but it's garbled.  On the MEGA-BAS, do you know which pin is High or Low?

**Reply by tagyoureit** [reply]
Check from the command line.  You can install the [mega-bas](https://github.com/SequentMicrosystems/megabas-rpi/blob/master/python/README.md) command line tools. 

When you run the following command the important part is that mode=0.  That is rs485 passthrough.  The speed settings you see relate to mode=1 which is modbus.

```
pi@nixie:~ $ megabas 0 rs485rd
<mode> <baudrate> <stopbits> <parity> <add> 0 38400 1 0 1
```

`/dev/ttyAMA0` is for the video display ports and not applicable.  Just ignore them for this application.

**Reply by orrious** [reply]
```@raspberrypi:~/megabas-rpi $ megabas 0 rs485rd
<mode> <baudrate> <stopbits> <parity> <add> 0 38400 1 0 1
```
This looks good to me.  So megabas is in passthrough.

When it stops raining, I'll take a USB UART out and connect directly to pins 8 and 10 on the RPi header.

**Reply by orrious** [reply]
For whatever reason the serial port I have to use is `/dev/ttyAMA0`.  I have it working though through the MEGA-BAS

**Reply by tagyoureit** [reply]
It looks like it depends on the hw.  RPi3 on /dev/ttyAMA0 is /dev/ttyS0 on RPi4.  https://raspberrypi.stackexchange.com/questions/45570/how-do-i-make-serial-work-on-the-raspberry-pi3-pizerow-pi4-or-later-models/45571#45571

**Comment by orrious** [comment, marked as answer]
I completely forgot about this command and is what helped diagnose.

`od -x < /dev/ttyAMA0` and `od -x < /dev/ttyS0`
